pkg/cloudprovider/types.go

```diff
@@ -215,11 +215,11 @@ func (i InstanceTypeOverhead) Total() v1.ResourceList {
 }
 
 // An Offering describes where an InstanceType is available to be used, with the expectation that its properties
-// may be tightly coupled (e.g. the availability of an instance type in some zone is scoped to a capacity type)
+// may be tightly coupled (e.g. the availability of an instance type in some zone is scoped to a capacity type) and
+// these properties are captured with labels in the Constraints map
 type Offering struct {
-	CapacityType string
-	Zone         string
-	Price        float64
+	Constraints map[string]string
```

Comment:
Can we add a comment calling out which constraints are explicitly required? For example, we require a capacity-type constraint for disruption.

Reply:
Yep, I think that we keep Zone and CapacityType as required.